portfolio/src/main/webapp/script2.js

```diff
@@ -60,28 +65,32 @@ function tabulateAnswers() {
     // if the radio is checked..
     if (choices[i].checked) {
       // add 1 to that choice's score
-      if (choices[i].value =cvals[i]) {
-        cscores[i] = cscores[i] + 1;
-        console.log('here!');
+
+      cdict[choices[i].value] = cdict[choices[i].value] + 1; 
+      
       }
     }
-  }
   // Find out which choice got the highest score.
-  var maxscore = Math.max.apply(Math,cscores);
+  console.log(cdict);
+  var vals = Object.keys(cdict).map(function(key){
+    return cdict[key]});
```

Comment:
Watch out for this since it'll make a copy of cdict and store it into vals.  It won't be a problem for smaller examples like this.  But when you get to bigger data sets, extra copies would mean using up more memory.

A more efficient approach would be to loop around cdict, and compare each value to collect the max (rather than using the Math.max method).